orchagent/routeorch.cpp

```diff
@@ -297,65 +182,20 @@ bool RouteOrch::addRoute(IpPrefix ipPrefix, IpAddresses nextHops)
             if (to_add)
             {
                 /* Randomly pick an address from the address set. */
+                set<IpAddress> next_hop_set = nextHops.getIpAddresses();
                 auto it = next_hop_set.begin();
                 advance(it, rand() % next_hop_set.size());
 
-                /*
-                 * Set the route's temporary next hop to be the randomly picked one.
-                 */
+                /* Set the route's temporary next hop to be the randomly picked one. */
```

Comment:
it is possible that the choosed next hop is not in the neighbor list yet. then you are going to fail to add this tmp route. can we choose one that is already in the neighbor list?